### PR TITLE
Jim / improve deriv-com performance

### PR DESCRIPTION
**Pull request by jim-deriv**
Changes:

-   ...

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [X] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

**Review by yashim-deriv — DISMISSED** [review]

**Review by ashraf-deriv — APPROVED** [review]

File: src/components/hooks/use-website-status.tsx [hunk 1]
```diff
@@ -44,25 +45,39 @@ export const useWebsiteStatusApi = () => {
...
 }
+
+const getAltWebsiteStatus = () => {
+    let website_status = null
+    const xhttp = new XMLHttpRequest()
+    xhttp.onload = function () {
+        const data = this.responseText.match(/(?:loc)=(.*)$/gm)[0].substring(4)
+        website_status = { clients_country: data.toLocaleLowerCase() }
+    }
+    xhttp.open('GET', 'https://www.cloudflare.com/cdn-cgi/trace', false)
```
Comment (habib-deriv) [inline src/components/hooks/use-website-status.tsx:80]: Let's keep the  url in constants file.

File: src/components/hooks/use-website-status.tsx [hunk 2]
```diff
@@ -44,25 +45,39 @@ export const useWebsiteStatusApi = () => {
...
     }, [website_status_api])
 
     return website_status_api
 }
+
+const getAltWebsiteStatus = () => {
+    let website_status = null
+    const xhttp = new XMLHttpRequest()
+    xhttp.onload = function () {
+        const data = this.responseText.match(/(?:loc)=(.*)$/gm)[0].substring(4)
```
Comment (habib-deriv) [inline src/components/hooks/use-website-status.tsx:77]: Assign the magic number 4 to a proper named variable

File: src/components/hooks/use-website-status.tsx [hunk 3]
```diff
@@ -44,25 +45,39 @@ export const useWebsiteStatusApi = () => {
...
+        if (!website_status_api) {
+            const binary_socket = BinarySocketBase.init()
+            binary_socket.onopen = () => {
+                binary_socket.send(JSON.stringify({ website_status: 1 }))
+            }
 
-        binary_socket.onmessage = (msg) => {
-            const response = JSON.parse(msg.data)
+            binary_socket.onmessage = (msg) => {
+                const response = JSON.parse(msg.data)
```
Comment (habib-deriv) [inline src/components/hooks/use-website-status.tsx:58]: Let's have a fallback when the `msg.data` is empty

File: src/components/hooks/use-website-status.tsx [hunk 4]
```diff
@@ -15,6 +15,7 @@ export const useWebsiteStatus = () => {
 
     useLayoutEffect(() => {
         setLoading(true)
+        setWebsiteStatus(!website_status ? getAltWebsiteStatus() : website_status)
```
Comment (habib-deriv) [inline src/components/hooks/use-website-status.tsx:18]: Let's revert this logic based on truthy value `website_status`

File: src/components/hooks/use-website-status.tsx (hunk 3 above)
Comment (jim-deriv) [reply src/components/hooks/use-website-status.tsx:58]: Hello @habib-deriv,

This was existing code, and in this case, they keep on trying to send the websocket connection until it's not null. This is what we are running away from. The state's initial value is set to null by default as well  therefore, the only fall back would be to hardcode an object containing a default client's country. Would that be fine?

Comment (habib-deriv) [reply src/components/hooks/use-website-status.tsx:58]: No worries then. Let's keep it as it is..

**Review by prince-deriv — APPROVED** [review]

**Pull request closed without merging**